File: core/src/main/java/com/hgianastasio/biblioulbrav2/core/search/exceptions/PageOutOfBoundsException.java
package com.hgianastasio.biblioulbrav2.core.search.exceptions;

import com.hgianastasio.biblioulbrav2.core.search.models.SearchResult;

/**
 * Created by heitorgianastasio on 4/25/17.
 */

public class PageOutOfBoundsException extends RuntimeException {
    SearchResult result;

    public PageOutOfBoundsException(SearchResult result) {
        this.result = result;
    }

    public PageOutOfBoundsException(String s, SearchResult result) {
        super(s);
        this.result = result;
    }
}

File: core/src/main/java/com/hgianastasio/biblioulbrav2/core/search/exceptions/PageOutOfBoundsException.kt
package com.hgianastasio.biblioulbrav2.core.search.exceptions

import com.hgianastasio.biblioulbrav2.core.search.models.SearchResult

/**
 * Created by heitorgianastasio on 4/25/17.
 */
class PageOutOfBoundsException : RuntimeException {
    var result: SearchResult

    constructor(result: SearchResult) {
        this.result = result
    }

    constructor(s: String?, result: SearchResult) : super(s) {
        this.result = result
    }
}